packages/frontend/src/Macadam.svelte

```diff
@@ -0,0 +1,237 @@
+<script lang="ts">
+import { onMount } from 'svelte';
+import { Button, EmptyScreen, ErrorMessage, FormPage, Input } from '@podman-desktop/ui-svelte';
+import Link from './lib/Link.svelte';
+import { faCheck, faTriangleExclamation } from '@fortawesome/free-solid-svg-icons';
+import { goToDiskImages } from './lib/navigation';
+import { cleanup } from '@testing-library/svelte';
+import DiskImageIcon from './lib/DiskImageIcon.svelte';
+import { bootcClient } from './api/client';
+import type { VmDetails } from '@crc-org/macadam.js';
+
+interface Props {
+  imageName: string;
+  imagePath: string;
+}
+let { imageName, imagePath }: Props = $props();
+
+// Variables
+let errorFormValidation = $state('');
+let createInProgress = $state(false);
+let createErrorMessage = $state('');
+let createSuccessMessage = $state('');
+let sshPrivateKeyLocation = $state('');
+let sshUsername = $state('');
+let virtualMachineName = $state('');
+
+// An array of strings to store the list of VM names to be used for validation
+let existingVMNames = $state<string[]>([]);
+
+// Validation to check that the image path ends with .raw or .qcow2
+$effect(() => {
+  if (imagePath && !(imagePath.endsWith('.raw') || imagePath.endsWith('.qcow2'))) {
+    errorFormValidation = 'Only raw or qcow2 file formats are supported. Please select a valid image file.';
+  } else {
+    errorFormValidation = '';
+  }
+});
+
+// Validation for the virtual machine name, each time it changes or existingVMNames is updated, make sure it does not match any existing VM names
+$effect(() => {
```

Comment:
`errorFormValidation` is updated via two $effects and I don't think there's a guarantee on which order they run in, plus they have different triggers. i.e. I think there's a risk that at some point one of these sets an error and then the other one clears `errorFormValidation`. I think it's safer to merge these, use two different error strings, or some other method.